content/livesync/how-livesync-works.textile

```diff
@@ -0,0 +1,95 @@
+
+h3(#receiving-messages). Receiving a message
+
+In the Models SDK, incoming messages on a channel trigger the @merge@ function, which integrates the message content into the existing state of your application.
+
+h3(#subscribing). Subscribing to changes
+
+In the Models SDK, you can subscribe to a model to receive its updated state whenever changes occur. This enables you to seamlessly integrate the updated model state into your user interface or react to changes in various ways.
+
+h3(#optimistic-updates). Optimistic updates
+
+@Optimistic@ updates to enhance UI responsiveness, allowing updates to be applied optimistically in the UI before being confirmed in the database. Each optimistic update is associated with a mutation ID, which enables tracking and matching with events received on the channel. This ID should be included in the mutation sent to your backend and recorded in the outbox table to confirm or reject the update. Unconfirmed optimistic updates will be rolled back by the Models SDK automatically.
+
+h2(#backend). Backend
+
+h3(#outbox-table). Database Outbox table
+
+The @outbox table@ serves as the equivalent of publishing to clients subscribed with Ably LiveSync. Automatically reacting to new records written to the outbox, the database connector publishes these records to clients. This approach ensures transactionality between database changes and records published to clients, enhancing data consistency and reliability.
+
+h3(#sequence-id). Sequence ID
+
+The Models SDK utilizes a sequence ID to ensure accurate matching of the model's state with the corresponding point in the change events stream obtained from Ably. This sequence ID, retrieved from your backend during the @sync@ function call, enables the Models SDK to synchronize the model's state effectively during the initial load process.
+
+<a href="@content/diagrams/what-is-livesync.png" target="_blank">
+  <img src="@content/diagrams/what-is-livesync.png" style="width: 100%" alt="What is LiveSync">
+</a>
+
+h2(#use-cases). Use-cases
+
+There are many possible problems that LiveSync can solve or simplify for projects. Below is a list of some of these:
+
+h3(#crm-systems). Customer relationship management systems
+
+A customer relationship management (CRM) system helps organistations to build customer relationships and improve processes to increase sales, improve customer service and manage customer data.
+
+A problem LiveSync can resolve within a customer relationship management (CRM) system, is on the individual customer page rendering their details or other models related to the customer on this page. An example of another model is orders by this customer. When any piece of this detail or their order is updated, this can be automatically picked up by LiveSync and update all frontend clients subscribed to this customer.
```

Comment:
This sounds a bit like it's describing the bits that livesync is more challenging to work with, i.e. having the same backend data represented in different models on the client, which requires multiple outbox writes (for each model).

See: https://docs.google.com/document/d/1Fj8Pm_qrC96RVA6rfycLF9p8c95b7yBoFuJWaPIDxcU/edit

>Models would not work well for the list pages or orders or customers, and would not work well for the relational aspects of the CRM (for example, it’d be hard to update all the individual ‘order’ models when a ‘customer’ changed their name, as each order would likely be its own model with it’s own channel, so the database would need to publish an event per-order-model).

Reply:
I think the problem here is though, that LiveSync for CRM is very specific on what it can help with. I've tried to not make it very clear but instead focusing on how on that part it can help. Do you have any recommendations on how to promote LiveSync for a CRM?

Reply:
Maybe focus on the collaborative form building aspect, rather than making it CRM-specific. E.g. if you have a screen open with a bunch of fields, you can see changes made by other users to those fields as they happen.